aws-samples/amazon-kinesis-video-streams-demos
# Python script to download long clips from Kinesis Video Streams

*Description of changes:*
* Python script to download a long clip from Kinesis Video Streams using the [`GetClip API`](https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_reader_GetClip.html).

*Testing:*
* Run on an environment where `ffmpeg` is not installed to verify `ffmpeg` check.
* Specify start_timestamp > end_timestamp to verify error handling.
* Specify negative chunk_size to verify error handling.
* Remove IAM permissions to verify error handling.
* Specify nonexistent stream name to verify error handling.
* Remove file write permissions to verify error handling.
* Download continuous 7-minute clip with 3-minute chunk size to confirm merging.
* Download continuous 7-minute clip with 1-minute chunk size to confirm merging.
* Fetch 1-minute clip with a 3-minute chunk size to confirm clip is fetched and no merging occurs.
* Fetch 2-hour clip with a section in the middle where I stopped uploading (no media present).
* Fetch 7-minute clip with a section in the end where I stopped uploading (no media present).
* Fetch 2-hour clip with a section at the beginning where I stopped uploading (no media present).
* Fetch 7-minute clip where no media was uploaded at all.
* Fetch 2-hour clip such that only one chunk had media to verify that no merging occurs when it all fits into a single chunk.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
